### DISCUSSION TITLE
Function signatue in wasm64

### DISCUSSION THREAD

**Discussion by tbuchs** [emscripten-core/emscripten#22135] [Q&A] [answered]
Hi, I try to port my code to wasm64 and have some issues with calling a thread on the main thread. I use the flag -sPROXY_TO_PTHREAD and need some work to be done of the main thread. I used to use `emscripten_sync_run_in_main_runtime_thread` and it worked perfectly fine. However, now with wasm64 I have the issue, that I always get the error of a wrong function signature.
The function looks like this: `    void (*ptr)(void*, unsigned char*, int) = &callback_method;` - what would be the right `EM_FUNC_SIG_` to use then?

I tried EM_FUNC_SIG_VPPI, but as far as I can see, this doesn't exist. Thanks for any help!

**Comment by sbc100** [comment, marked as answer]
Yes, we only added a few `EM_FUNC_SIG` variants with `P` in them.   This is because we tend to use the new `proxying.h` API for proxying work these days.  If you would like add a more `EM_FUNC_SIG` variants feel free to send a PR.

**Reply by tbuchs** [reply]
Interesting, is there a way to achieve the behavior of `emscripten_sync_run_in_main_runtime_thread` by using `proxying.h`?
Otherwise I gladly add a new variant - like always: thanks again for your great support :)

**Reply by sbc100** [reply]
Yes, if you want to switch to the API you would use `emscripten_proxy_get_system_queue()` to get the system queue and then you can use `emscripten_proxy_sync` or `emscripten_proxy_sync_with_ctx` to do the actual proxying.    

I will likely take a few more lines of code and since we don't use those type reflecting macros in the new API.. its more straight forward C.  However I think there is a C++ API that is more powerful.

**Reply by tbuchs** [reply]
Thanks for your help, this works perfectly!
